### ISSUE TITLE
Random restarts on external-scaler

### ISSUE THREAD

**Issue by huang-jy** [kedacore/http-add-on#630]
### Report

While experimenting with the scaler, I have noticed the external scaler seems to often restart during normal service, with a very cryptic message in stdout that makes no sense. During the downtime, the scaler does not scaleup or scaledown deployments, nor does it route traffic during this time, leading to timeouts.

### Expected Behavior

Service not to restart randomly and/or more informative message in output so it is easier to understand why it is having trouble

### Actual Behavior

After running a deployment that has multiple httpscaledobject definitions, the http-external-scaler will randomly decide to restart, outputting this very cryptic stack trace.

Since the stack trace is extremely large, I've put it in a gist:

https://gist.github.com/huang-jy/ba63113a579ba49846603185d6bc7b16

### Steps to Reproduce the Problem

The last service I deployed before this behaviour started uses Helm and a range loop to produce multiple consistent resources of the same template.

The service keda fronts is basically a simple Flask app that presents records from a Redis cluster in the form of an RSS feed, hence the naming

`keda.yaml`
```yaml
{{- $domain := .Values.ingress.domain }}
{{- $ingressClass := .Values.ingress.ingressclass }}
{{- $minreplicas := .Values.keda.minreplicas }}
{{- $maxreplicas := .Values.keda.maxreplicas }}
{{- $kedaNamespace := .Values.keda.kedanamespace }}

{{- if eq .Values.keda.enabled true }}
{{- range .Values.serverless.lines }}
---
kind: HTTPScaledObject
apiVersion: http.keda.sh/v1alpha1
metadata:
    name: {{ . }}
spec:
    host: {{ . }}.{{ $domain }} ## Should match the one in ingress
    targetPendingRequests: 200
    scaleTargetRef:
        deployment: rss-{{.}}
        service: rss-{{.}}
        port: 8080
    replicas:
        min: 0
        max: 4

---
## The ingress must go into the keda namespace
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: {{.}}
  namespace: {{ $kedaNamespace }}
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /
    kubernetes.io/ingress.class: {{ $ingressClass }}
spec:
  # ingressClassName: {{ $ingressClass }}
  rules:
  - host: {{ . }}.{{ $domain }} ## Should match the one in httpscaledobject
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: keda-add-ons-http-interceptor-proxy
            port:
              number: 8080

---
apiVersion: v1
kind: Service
metadata:
  annotations: {}
  labels: {}
  name: rss-{{.}}
spec:
  ports:
  - name: http
    port: 8080
    protocol: TCP
    targetPort: http #Must match a port block on the deployment spec containers template
  selector: 
    app: rss-{{.}}
  type: ClusterIP

{{- end }}
{{- end }}
```

`values.yaml` (NOTE: Only values applicable to the above are shown)
```yaml
ingress:
  domain: [HIDDEN].com
  ingressclass: nginx

keda:
  enabled: true
  minreplicas: 0
  maxreplicas: 4
  kedanamespace: keda

serverless:
  lines:
    - line1
    - line2
    - line3
    - line4
    - line5
    - line6
    - line7
    - line8
    - line9
    - line10
```

### Logs from KEDA HTTP operator

```
2023-03-13T12:13:29Z	ERROR	external_push_scaler	error	{"type": "ScaledObject", "namespace": "my-namespace", "name": "line1-app", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp 10.68.11.248:9090: i/o timeout\""}
github.com/kedacore/keda/v2/pkg/scalers.(*externalScaler).GetMetricSpecForScaling
	/workspace/pkg/scalers/external_scaler.go:147
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).GetMetricSpecForScalingForScaler
	/workspace/pkg/scaling/cache/scalers_cache.go:104
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).getScaledObjectState
	/workspace/pkg/scaling/scale_handler.go:561
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:228
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:167
2023-03-13T12:13:29Z	ERROR	external_push_scaler	error	{"type": "ScaledObject", "namespace": "my-namespace", "name": "line1-app", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp 10.68.11.248:9090: i/o timeout\""}
github.com/kedacore/keda/v2/pkg/scalers.(*externalScaler).GetMetricSpecForScaling
	/workspace/pkg/scalers/external_scaler.go:147
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).GetMetricSpecForScalingForScaler
	/workspace/pkg/scaling/cache/scalers_cache.go:113
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).getScaledObjectState
	/workspace/pkg/scaling/scale_handler.go:561
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:228
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:167
2023-03-13T12:13:29Z	ERROR	scale_handler	error getting metric spec for the scaler	{"scaledObject.Namespace": "my-namespace", "scaledObject.Name": "line1-app", "scaler": "externalPushScaler", "error": "got empty metric spec"}
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).getScaledObjectState
	/workspace/pkg/scaling/scale_handler.go:564
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:228
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:167
```

(Operator logs are full of similar messags from the other lines)

But I am not defining "line1-app" anywhere so I don't know why that's being requested

### What version of the KEDA HTTP Add-on are you running?

Chart keda-add-ons-http-0.4.1, App version 0.4.0      

### Kubernetes Version

1.23

### Platform

Google Cloud

### Anything else?

If I install the chart one item at a time, e.g. install with values:

```
serverless:
  lines:
    - line1
```

Then install it a second time using

```
serverless:
  lines:
    - line1
    - line2
```

And so on, Keda seems to be more stable, but I have no idea why, since I ultimately end up at the same goal, just taking far longer.

**Comment by tomkerkhove** [comment]
I think this is a concurrency issue and should be gone after @t0rr3sp3dr0 implemented #605.
Any other thoughts @t0rr3sp3dr0?

**Comment by huang-jy** [comment]
Let me double-check the values yaml when I'm back in the office tomorrow. I do recall we specify a custom values yaml for the install of the addon, and that may be the cause of the 0.3.0 version if it has been overridden in there.
